src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,251 @@
+        assertEquals(END_TIME, notificationsDb.getNotification(n.getNotificationId()).getEndTime());
+        assertEquals(END_TIME_DIFF, notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withEndTime(END_TIME_DIFF)
+                        .build()).getEndTime());
+        assertEquals(END_TIME_DIFF, notificationsDb.getNotification(n.getNotificationId()).getEndTime());
+
+        ______TS("success: single field - shown");
+        assertFalse(notificationsDb.getNotification(n.getNotificationId()).isShown());
+        assertTrue(notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withShown()
+                        .build()).isShown());
+        assertTrue(notificationsDb.getNotification(n.getNotificationId()).isShown());
+
+        ______TS("success: bulk update");
+        // We will try to revert back to the original attribute to test bulk update
+        // Note: Update of shown is not revertable so not to check it here
+        n = notificationsDb.getNotification(n.getNotificationId());
+        assertEquals(TITLE_DIFF, n.getTitle());
+        assertEquals(MESSAGE_DIFF, n.getMessage());
+        assertEquals(TYPE_DIFF, n.getType());
+        assertEquals(TARGET_USER_DIFF, n.getTargetUser());
+        assertEquals(START_TIME_DIFF, n.getStartTime());
+        assertEquals(END_TIME_DIFF, n.getEndTime());
+        assertTrue(n.isShown());
+        notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withTitle(TITLE)
+                        .withMessage(MESSAGE)
+                        .withType(TYPE)
+                        .withTargetUser(TARGET_USER)
+                        .withStartTime(START_TIME)
+                        .withEndTime(END_TIME)
+                        .build());
+
+        n = notificationsDb.getNotification(n.getNotificationId());
+        assertEquals(TITLE, n.getTitle());
+        assertEquals(MESSAGE, n.getMessage());
+        assertEquals(TYPE, n.getType());
+        assertEquals(TARGET_USER, n.getTargetUser());
+        assertEquals(START_TIME, n.getStartTime());
+        assertEquals(END_TIME, n.getEndTime());
+        assertTrue(n.isShown());
+    }
+
+    @Test
+    public void testDeleteNotification() throws Exception {
+        ______TS("silent deletion of non-existant notification");
+        notificationsDb.deleteNotification("non.existent");
+
+        NotificationAttributes n = createNewNotification();
+        ______TS("typical success case");
+        assertNotNull(notificationsDb.getNotification(n.getNotificationId()));
+        notificationsDb.deleteNotification(n.getNotificationId());
+        assertNull(notificationsDb.getNotification(n.getNotificationId()));
+
+        ______TS("silent deletion of the same key twice");
```

Comment:
_"silent deletion of the same ~~key~~ notification twice"_?

Reply:
Thank you for the comment. It will be updated in the next commit.